Introducing the Paketo Liberty Buildpack (#2138)
OpenLiberty/blogs: posts/2022-04-01-cloud-native-liberty-buildpack.adoc

== What is a buildpack?

A buildpack is a set of executables that inspects your application source code and creates a plan to build your application.  

Paketo buildpacks implement the link:https://buildpacks.io[Cloud Native Computing Foundation] buildpack specification to provide toolkits and workflows for building reproducible container images from source code.

The link:https://github.com/paketo-buildpacks/liberty[Paketo Liberty buildpack] provides the Open Liberty runtime to a workflow that produces an Open Container Initiative (OCI) image that can run just about anywhere.

== Why use the Paketo Liberty buildpack?

* You can build your application without creating a Dockerfile!  

> * Advanced Caching - Paketo buildpacks use built-in caching to improve performance so you can quickly rebuild your application by updating only the layers that have changed.  

Reviewer: ```suggestion
* Advanced Caching: Paketo buildpacks use built-in caching to improve performance so you can quickly rebuild your application by updating only the layers that have changed.  
```

> * Bill-of-Materials - Built-in software bill of materials (SBOM) support to provide insights into the contents of the application image.  

Reviewer: ```suggestion
* Bill-of-Materials: a built-in software bill of materials (SBOM) support provides insights into the contents of the application image.  
```

> * Minimal Application Image - Images only contain what is necessary.

Reviewer: ```suggestion
* Minimal Application Image: images contain only what is necessary.
```

> * Reproducability - Reproduces the same application image digest by re-running the build. 

Reviewer: ```suggestion
* Reproducibility: reproduce the same application image digest by re-running the build. 
```

> * Auto-detection - Images are built directly from application source. 

Reviewer: ```suggestion
* Auto-detection: images are built directly from application source. 
```

> * Rebasing - Instant updates of base images by patching the OS layer of your image without rebuilding your source code.  

Reviewer: ```suggestion
* Rebasing: instantly update base images without rebuilding your source code by patching the OS layer of your image. 
```

[source]
pack build myapp --env BP_JAVA_APP_SERVER=liberty

Your application is now transformed into an OCI image!

== Now what? 

> With your OCI image you can run your application locally:

Reviewer: ```suggestion
With your OCI image, you can run your application locally with the `docker run` command:
```

[source]
docker run --rm -p 9080:9080 myapp 

> or deploy your application to any Kubernetes-based platform such as link:https://www.redhat.com/en/technologies/cloud-computing/openshift[Red Hat OpenShift] using an link:https://github.com/OpenLiberty/open-liberty-operator[Open Liberty operator]

Reviewer: ```suggestion
or deploy your application to any Kubernetes-based platform, such as link:https://www.redhat.com/en/technologies/cloud-computing/openshift[Red Hat OpenShift], by using an link:https://github.com/OpenLiberty/open-liberty-operator[Open Liberty operator]
```
